console_test.go

```diff
@@ -15,48 +15,71 @@
 package main
 
 import (
+	"io/ioutil"
 	"os"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
 func TestConsoleFromFile(t *testing.T) {
+	assert := assert.New(t)
+
 	console := ConsoleFromFile(os.Stdout)
 
-	if console.File() == nil {
-		t.Fatalf("console file is nil")
-	}
+	assert.NotNil(console.File(), "console file is nil")
 }
 
 func TestNewConsole(t *testing.T) {
+	assert := assert.New(t)
+
 	console, err := newConsole()
-	if err != nil {
-		t.Fatalf("failed to create a new console: %s", err)
-	}
+	assert.NoError(err, "failed to create a new console: %s", err)
 	defer console.Close()
 
-	if console.Path() == "" {
-		t.Fatalf("console path is empty")
-	}
+	assert.NotEmpty(console.Path(), "console path is empty")
 
-	if console.File() == nil {
-		t.Fatalf("console file is nil")
-	}
+	assert.NotNil(console.File(), "console file is nil")
 }
 
 func TestIsTerminal(t *testing.T) {
+	assert := assert.New(t)
+
 	var fd uintptr = 4
-	if isTerminal(fd) {
-		t.Fatalf("Fd %d is not a terminal", fd)
-	}
+	assert.False(isTerminal(fd), "Fd %d is not a terminal", fd)
 
 	console, err := newConsole()
-	if err != nil {
-		t.Fatalf("failed to create a new console: %s", err)
-	}
+	assert.NoError(err, "failed to create a new console: %s", err)
 	defer console.Close()
 
 	fd = console.File().Fd()
-	if !isTerminal(fd) {
-		t.Fatalf("Fd %d is a terminal", fd)
-	}
+	assert.True(isTerminal(fd), "Fd %d is a terminal", fd)
+}
+
+func TestReadWrite(t *testing.T) {
+	assert := assert.New(t)
+
+	// write operation
+	f, err := ioutil.TempFile(os.TempDir(), ".tty")
+	assert.NoError(err, "failed to create a temporal file")
+
+	console := ConsoleFromFile(f)
+	assert.NotNil(console)
+	defer console.Close()
+
+	msgWrite := "hello"
+	l, err := console.Write([]byte(msgWrite))
+	assert.NoError(err, "failed to write message: %s", msgWrite)
+	assert.Equal(len(msgWrite), l)
+
+	console.master.Sync()
+	console.master.Seek(0, 0)
+
+	// Read opeartion
+	msgRead := make([]byte, len(msgWrite))
+	l, err = console.Read(msgRead)
+	assert.NoError(err, "failed to read message: %s", msgWrite)
+	assert.Equal(len(msgWrite), l)
+
+	os.Remove(f.Name())
```

Comment:
Oh, it's down here ;) Not a problem but I generally add a `defer` straight after the error-checking to create it. The result is the same though - we want the file to exist on error.